gradle-plugin/plugin/src/main/groovy/io/realm/gradle/Realm.groovy

```diff
@@ -1,13 +1,126 @@
 package io.realm.gradle
-
+import com.android.build.gradle.AppPlugin
+import com.android.build.gradle.LibraryPlugin
+import javassist.*
+import javassist.expr.ExprEditor
+import javassist.expr.FieldAccess
+import org.gradle.api.GradleException
 import org.gradle.api.Plugin
 import org.gradle.api.Project
+import org.gradle.api.file.ConfigurableFileTree
+import org.gradle.api.file.FileCollection
+import org.gradle.api.tasks.compile.JavaCompile
 
 class Realm implements Plugin<Project> {
 
     @Override
     void apply(Project project) {
-    	project.repositories.add(project.repositories.jcenter())
+        // Make sure the project is either an Android application or library
+        def isAndroidApp = project.plugins.withType(AppPlugin)
+        def isAndroidLib = project.plugins.withType(LibraryPlugin)
+        if (!isAndroidApp && !isAndroidLib) {
+            throw new GradleException("'android' or 'android-library' plugin required.")
+        }
+
+        // Configure the project
+        project.repositories.add(project.repositories.jcenter())
         project.dependencies.add('compile', 'io.realm:realm-android:0.80.3') // TODO: make version dynamic
```

Comment:
I had some discussions with Nabil about this... we will take it all on Thursday cause it's important and has dramatic consequences